src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProvider.java

```diff
@@ -14,22 +14,20 @@ public class HSMBookKeepingClientProvider {
     private static final Logger logger = LoggerFactory.getLogger(HSMBookKeepingClientProvider.class);
     private static final int MIN_SUPPORTED_VERSION = 2;
     private static final int MAX_SUPPORTED_VERSION = 3;
-    private final HSMClientProtocol hsmClientProtocol;
 
-    public HSMBookKeepingClientProvider(HSMClientProtocol protocol) {
-        this.hsmClientProtocol = protocol;
+    public HSMBookKeepingClientProvider() {
     }
 
-    public HSMBookkeepingClient getHSMBookKeepingClient() throws HSMClientException {
-        int version = getVersion();
+    public HSMBookkeepingClient getHSMBookKeepingClient(HSMClientProtocol protocol) throws HSMClientException {
+        int version = protocol.getVersion();
         logger.debug("[getHSMBookKeepingClient] version: {}", version);
         HSMBookkeepingClient bookkeepingClient;
         switch (version) {
             case 1:
                 throw new HSMUnsupportedVersionException("HSMBookKeepingClient doesn't exist for version %s " + version);
```

Comment:
```suggestion
                throw new HSMUnsupportedVersionException("HSMBookKeepingClient doesn't exist for version 1");
```
That was wrong, to use %s it should have used String.format. But anyway, we already it's version 1 since it's inside the case

Reply:
Resolved